WebRoot/public/login.jsp

```diff
@@ -17,6 +17,26 @@
 <%-- this checks and redirects to admin if need be --%>
 <zm:adminRedirect/>
 <app:skinAndRedirect />
+<c:if test="${zm:isDomainLoginPageEnabled()}">
+    <zm:getDomainInfo var="domainInfo" by="virtualHostname" value="${zm:getServerName(pageContext)}"/>
+    <c:choose>
+        <c:when test="${not empty domainInfo}">
+            <c:set var="zimbraDomainLoginPagePath" value="${domainInfo.attrs.zimbraDomainLoginPagePath}" />
+            <c:choose>
+              <c:when test="${not empty zimbraDomainLoginPagePath}" >
+                <jsp:forward page="${zimbraDomainLoginPagePath}" />
+              </c:when>
+              <c:otherwise>
+                <jsp:forward page="${domainInfo.attrs.zimbraDomainLoginPageFallbackPath}" />
+              </c:otherwise>
+            </c:choose>
+        </c:when>
+        <c:otherwise>
+            <jsp:forward page="${domainInfo.attrs.zimbraDomainLoginPageErrorPath}" />
```

Comment:
I noticed that zimbraDomainLoginPageErrorPath must be globalConfig. If DomainInfo cannot be fetched, it also cannot be fetched.
I will update the PRs.

Reply:
* updated PRs on zm-mailbox and zm-taglib.
* changed to fetch zimbraDomainLoginPageErrorPath in `globalConfig` here